lang/python/CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+if(POLICY CMP0078)
+    cmake_policy(SET CMP0078 NEW)
+endif()
+
+# UseSWIG: Policy option to honor SWIG_MODULE_NAME.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
+
+find_package(SWIG REQUIRED)
+include(${SWIG_USE_FILE})
+
+set(python_libs)
+set(python_version)
+if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "3.12.0")
+    # This method of finding python libs has been deprecated since version 3.12.
+    # If we are running with a greater CMake version, opt to use the Python3 package.
+    set(Python_ADDITIONAL_VERSIONS 3.9 3.8 3.7 3.6 3.5)
+    find_package(PythonInterp REQUIRED)
+    find_package(PythonLibs REQUIRED)
+    include_directories(${PYTHON_INCLUDE_DIRS})
+    set(python_libs ${PYTHON_LIBRARIES})
+    set(python_version ${PYTHON_VERSION_STRING})
+else()
+    find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+    include_directories(${Python3_INCLUDE_DIRS})
+    set(python_libs ${Python3_LIBRARIES})
+    set(python_version ${Python3_VERSION})
+endif()
+
+# Manually specify the interface name '_wiredtiger' over a generated name (for backwards compatibility).
+list(APPEND swig_flags "-interface;_wiredtiger")
+
+list(APPEND swig_flags "-DPY_MAJOR_VERSION=${VERSION_MAJOR}")
+list(APPEND swig_flags "-threads")
+list(APPEND swig_flags "-O")
+list(APPEND swig_flags "-nodefaultctor")
+list(APPEND swig_flags "-nodefaultdtor")
+
+# Pass in our generated include directories.
+list(APPEND swig_flags "-I${CMAKE_BINARY_DIR}/include")
+list(APPEND swig_flags "-I${CMAKE_BINARY_DIR}/config")
+
+set(CMAKE_SWIG_FLAGS ${swig_flags})
+
+swig_add_library(_wiredtiger
+  TYPE SHARED
+  LANGUAGE python
+  SOURCES ${CMAKE_CURRENT_LIST_DIR}/wiredtiger.i
+)
+
+# Capture the target name generated by 'swig_add_library'. On older versions of cmake (< 3.13),
+# CMP0078 policy is default set to OLD (leading to a potentially different target name).
+set(swig_wt_target ${SWIG_MODULE__wiredtiger_REAL_NAME})
+
+# Depending on the chosen compiler (clang or gcc), a slightly different set of compiler flags
+# will be used.
+set(swig_c_flags)
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "AppleClang")
+    set(swig_c_flags
+        -Wno-pointer-sign
+        -Wno-ignored-qualifiers
+        -Wno-incompatible-pointer-types
+        -Wno-int-conversion
+        -Wno-sign-conversion
+    )
+else()
+    set(swig_c_flags
+        -Wno-discarded-qualifiers
+        -Wno-incompatible-pointer-types
+        -Wno-int-conversion
+        -Wno-sign-conversion
+    )
+endif()
+list(APPEND swig_c_flags -I${CMAKE_SOURCE_DIR})
+list(APPEND swig_c_flags -I${CMAKE_BINARY_DIR}/config)
+if((${SWIG_VERSION} VERSION_LESS 4) AND (${python_version} VERSION_GREATER_EQUAL 3.7))
+    # Older versions of SWIG (<= 3) will generate deprecated python3 buffer protocols (being marked as
+    # deprecated in Python 3.7+). Supress these warnings if we end up with an older SWIG compiler
+    # and newer python3 version.
```

Comment:
It's more natural to me the comments are put in front of the code logic (i.e. the if statement). I fully understand if you prefer to keep the current style :)

Reply:
No problem! I have not strong style preferences so I adjusted the comment to be above the if statement :)